piksi_tools/bootload.py

```diff
@@ -81,11 +81,16 @@ def wait_for_handshake(self, timeout=None):
     if timeout is not None:
       t0 = time.time()
     self.handshake_received = False
+    expire = time.time() + 15.0
+    self.link.send(SBP_MSG_RESET, "")
     while not self.handshake_received:
       time.sleep(0.1)
       if timeout is not None:
         if time.time()-timeout > t0:
           return False
+      if time.time() > expire:
+        expire = time.time() + 15.0
+        self.link.send(SBP_MSG_RESET, "")
```

Comment:
Makes sense to me - I updated the patch. `bootloader.handshake()` seems like a good description - handshake as a verb.